### ISSUE TITLE
[C#] Change CSComponent's FieldValues to use the field name instead of the hash of that name

### ISSUE THREAD

**Issue by mattbenic** [AtomicGameEngine/AtomicGameEngine#1431]
Currently the fieldvalues in CSComponent use the hash of a field's name to identify the field. This makes it much harder to work directly within the scene or prefab text when not working in the editor. This it also relevant when doing diffs between scenes or prefabs and trying to make sense of those changes. 

This should be made backwards compatible by supporting _reading_ hashes or names, but _always writing_ names.

**Comment by JoshEngebretson** [comment]
We can automatically register significant strings to a reverse lookup, for things like serialization fields and then use the string value instead of hash when writing the variant map.  This shouldn't be a ton of code and would be a significant improvement for development.  There is another issue similar to this I believe, we'll have to send out a search party for it 🚁⛰

**Comment by JoshEngebretson** [comment]
Ok, this should be a go once #1437 lands, resaving the scene should switch all fields in the serialized variant map to use the name instead of hash attribute.

Please lemme know if you run into any issues, btw, I am way too fond of making little emoji scenes, so here's one :) 🐐⛰

**Comment by mattbenic** [comment]
Oh awesome :) I'll probably do an upstream update next week, I'll let you know how it goes then
